02_bubblesort/main.go

```diff
@@ -0,0 +1,33 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+func bubble(items []int) []int {
+	n := len(items)
+	sorted := false
+
+	for !sorted {
+		swapped := false
+		for i := 0; i < n-1; i++ {
+			if items[i] > items[i+1] {
+				items[i+1], items[i] = items[i], items[i+1]
+				swapped = true
+			}
+		}
+		if !swapped {
```

Comment:
Directly use the `swapped` flag in for loop. When `swapped` is false the array is sorted.